redis/redis.go

```diff
@@ -181,6 +182,56 @@ func (c *RedisStore[V]) BatchSetEx(ctx context.Context, keys []string, values []
 	return nil
 }
 
+func (c *RedisStore[V]) GetEx(ctx context.Context, key string) (V, time.Duration, bool, error) {
+	var out V
+	var ttl time.Duration
+
+	_, err := c.client.TxPipelined(ctx, func(pipe redis.Pipeliner) error {
+		getVal := pipe.Get(ctx, key)
+		getTTL := pipe.TTL(ctx, key)
+
+		_, err := pipe.Exec(ctx)
+		if errors.Is(err, redis.Nil) {
+			return err
+		}
+
+		if err != nil {
+			return fmt.Errorf("exec: %w", err)
+		}
+
+		ttl, err = getTTL.Result()
+		if err != nil {
+			return fmt.Errorf("TTL command failed: %w", err)
+		}
+
+		if ttl == -1 {
+			ttl = time.Duration(math.MaxInt64)
+		}
+
+		data, err := getVal.Bytes()
+		if err != nil {
+			return fmt.Errorf("get bytes: %w", err)
+		}
+
+		out, err = deserialize[V](data)
+		if err != nil {
+			return fmt.Errorf("deserialize: %w", err)
+		}
+
+		return nil
+	})
+
+	if errors.Is(err, redis.Nil) {
+		return out, ttl, false, nil
+	}
+
+	if err != nil {
+		return out, ttl, false, fmt.Errorf("GetEx: %w", err)
+	}
```

Comment:
IMHO it is more readable, when you process the conditions for error in branch, instead of multiple if statements
```suggestion
	if err != nil {
		if errors.Is(err, redis.Nil) {
			return out, ttl, false, nil
		}
		
		return out, ttl, false, fmt.Errorf("GetEx: %w", err)
	}
```